docs: Add getting started docs for Ingress (#19760)
cilium-team/cilium: Documentation/gettingstarted/servicemesh/envoy-custom-listener.rst

.. only:: not (epub or latex or html)

    WARNING: You are looking at unreleased Cilium documentation.
    Please use the official rendered version released here:
    https://docs.cilium.io

.. _gs_envoy_custom_listener:

*******************
L7 Path Translation
*******************

This example replicates the Prometheus metrics listener which is

> already available via Cilium Agent ``--proxy-prometheus-port`` command

Reviewer: I think removing "Cilium Agent" is fine, because it's implied (feel free to add it back if you disagree). A definite article is needed though. 
```suggestion
already available via the command line option ``--proxy-prometheus-port``.
```

> line option. So the point of this example is not to add new

Reviewer: ```suggestion
So the point of this example is not to add new
```

functionality, but to show how a feature that previously required
Cilium Agent code changes can be implemented with the new Cilium Envoy
Config CRD.

Apply Example CRD
=================

This example adds a new Envoy listener ``envoy-prometheus-metrics-listener``

> on the standards Prometheus port (e.g. ``9090``) to each Cilium node, translating

Reviewer: ```suggestion
on the standard Prometheus port (e.g. ``9090``) for each Cilium node, translating
```

.. only:: not (epub or latex or html)

    WARNING: You are looking at unreleased Cilium documentation.
    Please use the official rendered version released here:
    https://docs.cilium.io

.. _gs_envoy_custom_listener:

*******************
L7 Path Translation
*******************

This example replicates the Prometheus metrics listener which is
already available via Cilium Agent ``--proxy-prometheus-port`` command
line option. So the point of this example is not to add new
functionality, but to show how a feature that previously required
Cilium Agent code changes can be implemented with the new Cilium Envoy
Config CRD.

Apply Example CRD
=================

This example adds a new Envoy listener ``envoy-prometheus-metrics-listener``
on the standards Prometheus port (e.g. ``9090``) to each Cilium node, translating
the default Prometheus metrics path ``/metrics`` to Envoy's Prometheus metrics path
``/stats/prometheus``.

Apply this Cilium Envoy Config CRD:

.. parsed-literal::

    $ kubectl apply -f \ |SCM_WEB|\/examples/kubernetes/servicemesh/envoy/envoy-prometheus-metrics-listener.yaml

> This version of the CEC CRD is Cluster-scoped, (i.e., not namespaced),

Reviewer 2: I don't think that `CEC` as an acronym was ever introduced. Maybe it should just always expand to ``` ``CiliumEndpointConfig`` ```? (Same with the other reference to `CEC`).

so the name needs to be unique in the cluster, unless you want to
replace a CRD with a new one.

.. Note::

    Note that these Envoy resources are not validated by k8s at all, so
    any errors in the Envoy resources will only be seen by the Cilium
    Agent observing these CRDs. This means that ``kubectl apply`` will
    report success, while parsing and/or installing the resources to the
    node-local Envoy instance may have failed. Currently the only way of
    verifying this is by observing Cilium Agent logs for errors and

>     warnings.

Reviewer 2: Probably these warnings should be in a dedicated file and included from the `envoy-traffic-management.rst` file as well, where that file is configuring CEC.

Reviewer 2: Eh, then again if `envoy-traffic-management.rst` is not getting the user to configure anything then maybe this one isn't as important to include over on that page since it's less likely something gets misconfigured.

> 

Reviewer 2: Also, what happens if the user configures multiple CEC? What happens if they configure Ingress and their own CEC? Are these supported? If not, these should also be called out in the limitations like the validation concern above.
